### DISCUSSION TITLE
Can't reopen telnet session

### DISCUSSION THREAD

**Discussion by chekhovana** [JSBSim-Team/jsbsim#1074] [Q&A]
Hello,

I'm facing the following problem: first telnet connection is established successfully:

```
$ telnet localhost 1137
Trying 127.0.0.1...
Connected to localhost.
Escape character is '^]'.
Connected to JSBSim server
JSBSim> quit
Closing connection
JSBSim> Connection closed by foreign host.
```
but the second attempt hangs without expected response, the output is:
```
$ telnet localhost 1137
Trying 127.0.0.1...
Connected to localhost.
Escape character is '^]'.
```

and nothing else. Is that the intended behaviour? I tried to skim through the source code and it seems that socket isn't closed completely on `quit` command. Adding extra line to `FGfdmSocket::Close` method fixes the issue:

original version:
https://github.com/JSBSim-Team/jsbsim/blob/b9bfc2350e0e00e53246b65979cc18df2922a88d/src/input_output/FGfdmSocket.cpp#L316-L323
modified version:
```c++
void FGfdmSocket::Close(void)
{
#ifdef _WIN32
  closesocket(sckt_in);
#else
  close(sckt_in);
#endif
  sckt_in = INVALID_SOCKET; // added line
}
```

But I'm not sure this is correct thing to do, as I'm extremely unexperienced both in c++ an sockets.

Any help would be appreciated.

**Comment by bcoconni** [comment]
Hi @chekhovana,

Indeed the method `FGfdmSocket::Receive` will not open a new socket if `sckt_in != INVALID_SOCKET`:
https://github.com/JSBSim-Team/jsbsim/blob/3c7582e3b1266eef8d4f0d66b0dc29a8643fe51d/src/input_output/FGfdmSocket.cpp#L245-L264
So your suggestion looks good to me. Would you mind opening a pull request to update the code ?
Thanks for your contribution.

**Reply by chekhovana** [reply]
Hello @bcoconni

Thanks for your response, here's my pull request: [https://github.com/JSBSim-Team/jsbsim/pull/1078](https://github.com/JSBSim-Team/jsbsim/pull/1078)
